From d54a743d6ae2826564a87add1225367285eff420 Mon Sep 17 00:00:00 2001
From: Jaeyoon Jung <jaeyoon.jung@lge.com>
Date: Mon, 28 Jan 2019 19:22:06 +0900
Subject: [PATCH] QQuickItemView: Remove redundant refill when layout

A refill() call after fixupPosition() in layout() is redundant. Even
worse it may cause a glitch by the re-construction of all visible items.

Change-Id: I08c3180eb2c2c79973bad7ada29d6fa956f5a3a3
Upstream-Status: Pending
---
 src/quick/items/qquickitemview.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/quick/items/qquickitemview.cpp b/src/quick/items/qquickitemview.cpp
index 164fff0fa..8864f9c5d 100644
--- a/src/quick/items/qquickitemview.cpp
+++ b/src/quick/items/qquickitemview.cpp
@@ -1871,10 +1871,8 @@ void QQuickItemViewPrivate::layout()
     markExtentsDirty();
     updateHighlight();
 
-    if (!q->isMoving() && !q->isFlicking() && !movingFromHighlight()) {
+    if (!q->isMoving() && !q->isFlicking() && !movingFromHighlight())
         fixupPosition();
-        refill();
-    }
 
     updateHeader();
     updateFooter();
